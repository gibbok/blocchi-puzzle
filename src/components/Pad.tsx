import React from 'react';
import styled from 'styled-components';
import { wood } from '../assets/images';

const PadStyled = styled.div`
  position: relative;
  width: 8rem;
  height: 8rem;
  border-radius: 4rem;
  font-size: 5rem;
  font-family: sans-serif;
  box-shadow: inset 0.1rem 0.1rem 0.3rem 0 rgba(0, 0, 0, 0.4);
  transform: rotate(90deg);
  clip-path: circle(50% at 50% 50%);
  border: 0.2rem solid rgba(0, 0, 0, 0.1);
  box-shadow: 10px 10px 5px 0px rgba(0, 0, 0, 0.12);
`;

const PadBackground = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-image: url(${wood});
  clip-path: circle(50% at 50% 50%);
`;

const Shadow = styled.div`
  filter: drop-shadow(-0.2rem 0.3rem 0.3rem rgba(50, 50, 0, 0.4));
`;

type Props = Readonly<{
  icon: React.ReactNode;
  onClick: () => void;
}>;

export function Pad({ icon, onClick }: Props) {
  return (
    <Shadow>
      <PadStyled onClick={onClick}>
        <PadBackground>{icon}</PadBackground>
      </PadStyled>
    </Shadow>
  );
}
