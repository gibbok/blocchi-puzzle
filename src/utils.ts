import { Tetro, Board } from './types';

export const logger = (x: Tetro | Board) => {
  console.log(x.reduce((acc, v) => `${acc} ${v.concat()} \n`, ''));
};
