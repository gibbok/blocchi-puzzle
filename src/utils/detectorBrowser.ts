import { detect } from 'detect-browser';

const getPathRelease = (version: string) => version.split('.')[0];

export const detectorBrowser = (): boolean => {
  const browser = detect();
  if (browser && browser.name && browser.version) {
    const name = browser.name;
    const version = browser.version;
    const path = Number(getPathRelease(version));

    switch (name) {
      case 'chrome':
      case 'firefox':
        return path > 80;
      case 'safari':
        return path > 13;
      case 'opera':
        return path > 62;
      default:
        return false;
    }
  } else {
    return false;
  }
};
