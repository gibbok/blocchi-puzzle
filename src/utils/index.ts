export * from './data.support.test';
export * from './logger';
