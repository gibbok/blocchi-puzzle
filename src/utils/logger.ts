import { Tetro, Board } from '../game/types';

export const logger = (x: Tetro | Board): void => {
  console.log(x.reduce((acc, v) => `${acc} ${v.concat()} \n`, ''));
};
