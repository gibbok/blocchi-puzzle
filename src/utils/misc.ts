import { IO, io } from 'fp-ts/lib/IO';
import { randomInt } from 'fp-ts/lib/Random';

export const getRandomValueFromStringEnum = <T>(strEnum: T): IO<T[keyof T]> => {
  const rndInt = randomInt(0, Object.keys(strEnum).length - 1)();
  const rndValue = Object.keys(strEnum)[rndInt] as keyof T;
  return io.of(strEnum[rndValue]);
};
