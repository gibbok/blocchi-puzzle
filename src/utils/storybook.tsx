import React from 'react';
import GlobalStyle from '../components/GlobalStyle';
import styled from 'styled-components';
import { wood } from '../assets/images';

/**
 * Decorate a story with glocal styles.
 * @param story Story
 */
export const globalStylesDecorator = (story: () => unknown): JSX.Element => (
  <React.Fragment>
    <GlobalStyle />
    {story()}
  </React.Fragment>
);

const Bkg = styled.div`
  background-image: url(${wood});
  width: 100vw;
  height: 100vh;
`;

/**
 * Decorate a story with a page background.
 * @param story Story
 */
export const pageBackground = (story: () => React.ReactNode): JSX.Element => <Bkg>{story()}</Bkg>;
