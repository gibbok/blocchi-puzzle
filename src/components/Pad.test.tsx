import React from 'react';
import { Pad } from './Pad';
import renderer from 'react-test-renderer';
import { noop } from '../utils';

it('should render Pad', () => {
  const tree = renderer.create(<Pad icon="close" onClick={noop} />).toJSON();
  expect(tree).toMatchSnapshot();
});
