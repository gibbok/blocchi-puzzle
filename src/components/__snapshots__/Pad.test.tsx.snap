// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Pad 1`] = `
<div
  className="sc-AxhUy ehyRyD"
>
  <div
    className="sc-AxirZ lcEoVe"
    onClick={[Function]}
  >
    <div
      className="sc-AxiKw cWlzsq"
    >
      close
    </div>
  </div>
</div>
`;
